### note: change all of the paths to your correct path when loading in data

In [1]:
import numpy as np
import pandas as pd
import matplotlib as mpl
import matplotlib.pyplot as plt
import scipy as scp
from scipy import stats
import statsmodels.api as sm
import statsmodels.formula.api as smf
import sklearn
from sklearn.linear_model import LinearRegression
import os
import Levenshtein

In [2]:
pd.options.display.max_rows = 600
pd.options.display.max_columns = 85

In [3]:
home_tmp = pd.read_csv('~/Dropbox/CDS-2019-AlbanyHub/Raw-Data/mismatched_home.csv', header = None)
cdbg_tmp = pd.read_csv('~/Dropbox/CDS-2019-AlbanyHub/Raw-Data/mismatched_cdbg.csv', header = None)
mismatched_home = home_tmp.iloc[:,0]
mismatched_cdbg = cdbg_tmp.iloc[:,0]

In [4]:
mismatched_home.shape[0] + mismatched_cdbg.shape[0]

99

### need to drop untitled column in each

In [3]:
df_full_utilities = pd.read_csv('~/Dropbox (Amherst College)/CDS-2019-AlbanyHub/ToDatabase/TotalUtilities.csv')

/anaconda3/envs/cds/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3049: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [4]:
s_addresses = pd.Series(df_full_utilities['Premise Address'].unique())

/anaconda3/envs/cds_env/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3049: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
/anaconda3/envs/cds_env/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3049: DtypeWarning: Columns (4,9) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [6]:
s_addresses[s_addresses=="529 COCHRAN AVE"]

Series([], dtype: object)

In [6]:
list1 = []
list2 = []
for df_u in df_utilities_list:
    list1.append(df_u['Premise Address'])
for series_u in list1:
    for row in series_u:
        list2.append(row)
series_utilities = pd.Series(list2)

In [10]:
s_utilities = pd.Series(series_utilities.unique())

NameError: name 'series_utilities' is not defined

In [12]:
xlsx = pd.ExcelFile('~/Dropbox/CDS-2019-AlbanyHub/Raw-Data/All CED Funded Projects 2007-2017.xlsx')
df1_original = pd.read_excel(xlsx, 'HOME Funded Projects')
df2_original = pd.read_excel(xlsx, 'CDBG Funded Projects')
df_original = pd.concat([df1_original, df2_original])
df_original.index = range(850)

In [23]:
df_houses = pd.read_csv('~/Dropbox (Amherst College)/CDS-2019-AlbanyHub/ToDatabase/TotalHome.csv')

### helper functions

In [8]:
def lev_list(address_list, address, i):
    temp_list = []
    temp_list_u = []
    for u_address in address_list:
        if Levenshtein.distance(address, u_address.upper()) <= i:
            temp_list.append(address)
            temp_list_u.append(u_address.upper())
    temp_list_series = pd.Series(temp_list)
    temp_list_u_series = pd.Series(temp_list_u)
    return temp_list_u_series

In [10]:
def space_fix_utilities(s_util):
    s_util_to_series = []
    for address in s_util:
        full_address = address.split()
        length = len(full_address)
        word = full_address[0] + "  "
        for i in range(1, length-1):
            word += full_address[i] + " "
        word += " " + full_address[length-1]
        word = word.strip()
        s_util_to_series.append(word)
    return pd.Series(s_util_to_series)

In [11]:
def construct_fixed_addresses(list_series):
    tmp_to_series = []
    for (apt, dup, sno, snm, stp) in zip(list_series[0],list_series[1],list_series[2],
                                         list_series[3],list_series[4]):
        word = ""
        if sno != "NA":
            word += sno + " "
        if snm != "NA":
            word += snm + " "
        if stp != "NA":
            word += stp
        tmp_to_series.append(word.strip())
    return pd.Series(tmp_to_series)

In [12]:
def gen_list_none(address_list, mismatched, i):
    tmp_list = []
    for address in mismatched:
        if address == "UNKNOWN ADDRESS" or address == "SUPPRESSED ADDRESS":
            continue
        if lev_list(address_list, address, i).shape[0] == 0:
            tmp_list.append(address)
    return pd.Series(tmp_list)

### Read in more data

In [13]:
# s_utilities = space_fix_utilities(s_utilities)
df_parsehome = pd.read_csv('~/Dropbox/CDS-2019-AlbanyHub/Raw-Data/fixed_home_v02.csv')
df_parsecdbg = pd.read_csv('~/Dropbox/CDS-2019-AlbanyHub/Raw-Data/fixed_cdbg.csv')

In [14]:
s_parsehome = df_parsehome['Address']
s_parsecdbg = df_parsecdbg['Address']

In [15]:
def containsAPT(series):
    apt_list = []
    for address in series:
        if "APT" in address:
            apt_list.append(address)
    return pd.Series(apt_list)

In [16]:
def containsDUP(series):
    dup_list = []
    for address in series:
        full_address = address.split(' ')
        length = len(full_address)
        for i in range(length - 1):
            word = full_address[i]
            if len(word) == 1:
                if not ((word == 'N') or (word == 'E')
                or (word == 'S') or (word == 'W')):
                    dup_list.append(address)
    return pd.Series(dup_list)

In [17]:
def clean(series):
    tmp_list = []
    for address in series:
        full_address = address.split(' ')
        length = len(full_address)
        if any(char.isdigit() for char in full_address[length-1]):
            tmp_list.append(address)
        else:
            tmp_list.append("---")
    return pd.Series(tmp_list)

In [18]:
def splitter(series):
    apt_list = set(containsAPT(series))
    dup_list = set(containsDUP(series))
    sno_to_series = []
    snm_to_series = []
    stp_to_series = []
    apt_to_series = []
    dup_to_series = []

    for address in series:
        full_address = address.split()
        length = len(full_address)
        begin_index = 1
        end_index = length - 1
        #mutually exclusive apt and duplex
        if address == "UNKNOWN ADDRESS":
            sno_to_series.append("NA")
            snm_to_series.append("UNKNOWN ADDRESS")
            stp_to_series.append("NA")
            apt_to_series.append("NA")
            dup_to_series.append("NA")
            continue
        if address == "SUPPRESSED ADDRESS":
            sno_to_series.append("NA")
            snm_to_series.append("SUPPRESSED ADDRESS")
            stp_to_series.append("NA")
            apt_to_series.append("NA")
            dup_to_series.append("NA")
            continue
        if address in apt_list:
            apt_to_series.append(full_address[length-1])
            stp_to_series.append(full_address[length-3])
            end_index -= 2
        else:
            apt_to_series.append("NA")
            stp_to_series.append(full_address[length-1])
            
        if address in dup_list:
            dup_to_series.append(full_address[1])
            begin_index += 1
        else:
            dup_to_series.append("NA")
        sno_to_series.append(full_address[0])
        to_str = full_address[begin_index:end_index]
        word = ""
        for part in to_str:
            word += part + " "
        snm_to_series.append(word.strip())
    series_apt = pd.Series(apt_to_series)
    series_dup = pd.Series(dup_to_series)
    series_sno = pd.Series(sno_to_series)
    series_snm = pd.Series(snm_to_series)
    series_stp = pd.Series(stp_to_series)
    return (series_apt, series_dup, series_sno, series_snm, series_stp)

In [19]:
series_apt_home, series_dup_home, series_sno_home, series_snm_home, series_stp_home = splitter(s_parsehome)
series_apt_cdbg, series_dup_cdbg, series_sno_cdbg, series_snm_cdbg, series_stp_cdbg = splitter(s_parsecdbg)

In [20]:
home = [series_apt_home, series_dup_home, series_sno_home, series_snm_home, series_stp_home]
cdbg = [series_apt_cdbg, series_dup_cdbg, series_sno_cdbg, series_snm_cdbg, series_stp_cdbg]
for hom in home:
    print(hom.shape)
for cdb in cdbg:
    print(cdb.shape)

(286,)
(286,)
(286,)
(286,)
(286,)
(564,)
(564,)
(564,)
(564,)
(564,)


In [21]:
def checker(list_series):
    tmp_to_series = []
    for (apt, dup, sno, snm, stp) in zip(list_series[0],list_series[1],list_series[2],
                                         list_series[3],list_series[4]):
        word = ""
        if sno != "NA":
            word += sno + " "
        if dup != "NA":
            word += dup + " "
        word += snm + " "
        if stp != "NA":
            word += stp + " "
        if apt != "NA":
            word += "APT " + apt
        word = word.strip()
        tmp_to_series.append(word)
    return pd.Series(tmp_to_series)

In [22]:
s_parsehome[checker(home) != s_parsehome]

Series([], Name: Address, dtype: object)

In [23]:
df_all_addresses = pd.read_csv('~/Dropbox/CDS-2019-AlbanyHub/Processed-Data/all_addresses.csv')
df_all_addresses = df_all_addresses.drop('Unnamed: 0', axis = 1)
s_all_addresses = df_all_addresses['line1']

### Workflow

s_utilities is a list of all addresses that show up in the utilities data <br>
s_all_addresses is a list of all addresses in Albany <br>
df_original is a list of all housing projects <br>
lev_list takes in one of the two lists above, an address, and a threshold. <br>
If the address is within the threshold distance (string comparison) of an address in the two lists, it will print out.

In [24]:
mismatched_home

0             2620  S MADISON  ST
1               529  COCHRAN  AVE
2          2226  OLD DOMINION  DR
3             2701  DONCASTER  RD
4            250  BONNY VIEW  AVE
5            249  BONNY VIEW  AVE
6            248  BONNY VIEW  AVE
7            251  BONNY VIEW  AVE
8            253  BONNY VIEW  AVE
9            247  BONNY VIEW  AVE
10           254  BONNY VIEW  AVE
11           243  BONNY VIEW  AVE
12           245  BONNY VIEW  AVE
13           252  BONNY VIEW  AVE
14            2010  LAFAYETTE  CR
15              20  S JACKSON  ST
16               603  N HALEY  ST
17           1112  PEACHTREE  TER
18             2517  CARDINAL  DR
19            2006  LAFAYETTE  CR
20              818  WESTVIEW  DR
21              2501  REDWOOD  CR
22              933  RANDOLPH  DR
23            2201  CHAMPAGNE  DR
24              215  BROADWAY  ST
25              1003  HOLLAND  DR
26            2200  CHAMPAGNE  LN
27           1019  UNIVERSITY  DR
28            509  WESTOVER  BLVD
29            

In [25]:
mismatched_cdbg

0                         UNKNOWN ADDRESS
1                          400  EBONY  LN
2                    638  W HIGHLAND  AVE
3                         UNKNOWN ADDRESS
4                         UNKNOWN ADDRESS
5                         UNKNOWN ADDRESS
6                  230  S JACKSON ST  250
7                         UNKNOWN ADDRESS
8                         UNKNOWN ADDRESS
9                         UNKNOWN ADDRESS
10                       302  S CASON  ST
11                        UNKNOWN ADDRESS
12                     SUPPRESSED ADDRESS
13                     225  S JACKSON  ST
14                   1010  W LINCOLN  AVE
15    1401  E CAMPBELL ST/108 CARROLL  ST
16                     2572  CRESCENT  DR
17                     2572  CRESCENT  DR
18                       4835  IMPALA  LN
19                         507  EMILY  ST
20                    608  RESIDENCE  AVE
21                       614  ROADWAY  ST
22                       706  E TOWNE  RD
23                     SUPPRESSED 

### example (note the spacing difference between addresses in utilities and the full addr list)

In [14]:
lev_list(s_addresses, "529 COCHRAN AVE", 4)

0        829 CORN AVE
1        729 CORN AVE
2        509 CORN AVE
3        525 CORN AVE
4        524 CORN AVE
5        629 CORN AVE
6     520 COCHRAN AVE
7     606 COCHRAN AVE
8     518 COCHRAN AVE
9      527 COCHRAN ST
10     525 COCHRAN ST
11    521 COCHRAN AVE
12     519 COCHRAN ST
13    524 COCHRAN AVE
14    528 COCHRAN AVE
15      529 GOWAN AVE
16     529 COCHRAN ST
17     522 COCHRAN ST
18     526 COCHRAN ST
dtype: object

In [27]:
lev_list(s_all_addresses, "1010 W LINCOLN AVE", 4)

0      1310 W LINCOLN AVE
1      1308 W LINCOLN AVE
2      1306 W LINCOLN AVE
3      1304 W LINCOLN AVE
4      1302 W LINCOLN AVE
5      1300 W LINCOLN AVE
6      1230 W LINCOLN AVE
7      1228 W LINCOLN AVE
8      1226 W LINCOLN AVE
9      1224 W LINCOLN AVE
10     1224 W LINCOLN AVE
11     1222 W LINCOLN AVE
12     1220 W LINCOLN AVE
13     1218 W LINCOLN AVE
14     1218 W LINCOLN AVE
15     1216 W LINCOLN AVE
16     1212 W LINCOLN AVE
17     1208 W LINCOLN AVE
18     1206 W LINCOLN AVE
19     1204 W LINCOLN AVE
20     1202 W LINCOLN AVE
21     1200 W LINCOLN AVE
22     1126 W LINCOLN AVE
23     1124 W LINCOLN AVE
24     1122 W LINCOLN AVE
25     1120 W LINCOLN AVE
26     1118 W LINCOLN AVE
27     1116 W LINCOLN AVE
28     1114 W LINCOLN AVE
29     1112 W LINCOLN AVE
30     1110 W LINCOLN AVE
31     1108 W LINCOLN AVE
32     1106 W LINCOLN AVE
33     1104 W LINCOLN AVE
34     1400 W LINCOLN AVE
35     1402 W LINCOLN AVE
36     1404 W LINCOLN AVE
37     1512 W LINCOLN AVE
38     1510 

### So in the above example, 2620 S Madison St is found in the list of all addresses, but not found in the utilities data. Record this fact in the google doc. Then, look up the project type in the housing project data to also record what type of project it was. I'm looking to see if there is any trend with demolition or disposition projects not being found in utilities data.

### example (note the capitalization difference)

In [28]:
mismatched_cdbg

Parcel ID               Address    City State XY Coordinates  Plan Year  \
286       NaN   1008 East Residence  Albany    GA            NaN       2007   
328       NaN   801 W Residence Ave  Albany    GA            NaN       2007   
499       NaN  608 Residence Avenue  Albany    GA            NaN       2010   
786       NaN     1505 E. Residence  Albany    GA            NaN       2016   
816       NaN     1329 E. Residence  Albany    GA            NaN       2017   

     IDIS Project            Project  IDIS Activity ID Activity Status  \
286             7  Emergency Repairs               578       Completed   
328            15     Public Service               572       Completed   
499            13         Demolition               941       Completed   
786             5  Emergency Repairs              1470       Completed   
816            20  Emergency Repairs              1518       Completed   

    Program  Funded Amount  Draw Amount  Balance  
286    CDBG        5671.14      5671.14      0.0  
328    CDBG       18572.87     18572.87      0.0  
499    CDBG        3341.25      3341.25      0.0  
786    CDBG        5884.47      5884.47      0.0  
816    CDBG       11896.43     11896.43      0.0

### Tests

In [29]:
mismatched_home

0                         UNKNOWN ADDRESS
1                          400  EBONY  LN
2                    638  W HIGHLAND  AVE
3                         UNKNOWN ADDRESS
4                         UNKNOWN ADDRESS
5                         UNKNOWN ADDRESS
6                  230  S JACKSON ST  250
7                         UNKNOWN ADDRESS
8                         UNKNOWN ADDRESS
9                         UNKNOWN ADDRESS
10                       302  S CASON  ST
11                        UNKNOWN ADDRESS
12                     SUPPRESSED ADDRESS
13                     225  S JACKSON  ST
14                   1010  W LINCOLN  AVE
15    1401  E CAMPBELL ST/108 CARROLL  ST
16                     2572  CRESCENT  DR
17                     2572  CRESCENT  DR
18                       4835  IMPALA  LN
19                         507  EMILY  ST
20                    608  RESIDENCE  AVE
21                       614  ROADWAY  ST
22                       706  E TOWNE  RD
23                     SUPPRESSED 

In [20]:
lev_list(s_utilities, "2312  GLENEAGLE  DR", 1)

NameError: name 'construct_fixed_addresses' is not defined

In [21]:
mismatched_home

0             2620  S MADISON  ST
1               529  COCHRAN  AVE
2          2226  OLD DOMINION  DR
3             2701  DONCASTER  RD
4            250  BONNY VIEW  AVE
5            249  BONNY VIEW  AVE
6            248  BONNY VIEW  AVE
7            251  BONNY VIEW  AVE
8            253  BONNY VIEW  AVE
9            247  BONNY VIEW  AVE
10           254  BONNY VIEW  AVE
11           243  BONNY VIEW  AVE
12           245  BONNY VIEW  AVE
13           252  BONNY VIEW  AVE
14            2010  LAFAYETTE  CR
15              20  S JACKSON  ST
16               603  N HALEY  ST
17           1112  PEACHTREE  TER
18             2517  CARDINAL  DR
19            2006  LAFAYETTE  CR
20              818  WESTVIEW  DR
21              2501  REDWOOD  CR
22              933  RANDOLPH  DR
23            2201  CHAMPAGNE  DR
24              215  BROADWAY  ST
25              1003  HOLLAND  DR
26            2200  CHAMPAGNE  LN
27           1019  UNIVERSITY  DR
28            509  WESTOVER  BLVD
29            

In [22]:
mismatched_cdbg

0                         UNKNOWN ADDRESS
1                          400  EBONY  LN
2                    638  W HIGHLAND  AVE
3                         UNKNOWN ADDRESS
4                         UNKNOWN ADDRESS
5                         UNKNOWN ADDRESS
6                  230  S JACKSON ST  250
7                         UNKNOWN ADDRESS
8                         UNKNOWN ADDRESS
9                         UNKNOWN ADDRESS
10                       302  S CASON  ST
11                        UNKNOWN ADDRESS
12                     SUPPRESSED ADDRESS
13                     225  S JACKSON  ST
14                   1010  W LINCOLN  AVE
15    1401  E CAMPBELL ST/108 CARROLL  ST
16                     2572  CRESCENT  DR
17                     2572  CRESCENT  DR
18                       4835  IMPALA  LN
19                         507  EMILY  ST
20                    608  RESIDENCE  AVE
21                       614  ROADWAY  ST
22                       706  E TOWNE  RD
23                     SUPPRESSED 

In [34]:
s_fixed_home

0                   1707 EDGERLY AVE
1                   703 DARTMOUTH LN
2                 1115 W WADDELL AVE
3                   313 FORRESTAL LN
4                       1125 JOEL DR
5                     1519 MALONE DR
6                     719 ANDOVER LN
7                305 S SHADOWLAWN DR
8                      1662 SUNNY LN
9                    511 MOULTRIE RD
10                     1100 SWIFT ST
11              1009 E ROOSEVELT AVE
12                1118 W WHITNEY AVE
13                      431 EBONY LN
14                      529 EBONY LN
15                      535 EBONY LN
16                      539 EBONY LN
17                      545 EBONY LN
18                      559 EBONY LN
19                      563 EBONY LN
20                      531 EBONY LN
21                      537 EBONY LN
22                      543 EBONY LN
23                      555 EBONY LN
24                1403 GREENBRIER CT
25                    1802 SHARON DR
26                   533 AMERICAN LN
27                   539 AMERICAN LN
28                   1602 ESTELLE ST
29                    456 GAINES AVE
30                     621 BURKE AVE
31                    1910 JONES AVE
32                    514 EUGEMAR DR
33                    456 GAINES AVE
34                    460 GAINES AVE
35                    460 GAINES AVE
36               616 INDIAN CREEK DR
37     1005 MARTIN LUTHER KING JR DR
38                    1411 OWENS AVE
39              2230 S WASHINGTON ST
40               1006 S JEFFERSON ST
41                    402 COTTON AVE
42                    1403 OWENS AVE
43                  426 ROBINSON AVE
44                   1212 S DAVIS ST
45                   1203 HAMPTON CT
46                   1209 HAMPTON CT
47                     118 ALMOND DR
48                   1305 HAMPTON CT
49                   1305 HAMPTON CT
50                   509 AMERICAN LN
51                   519 AMERICAN LN
52                   1324 MOBILE AVE
53                 1004 AMSTERDAM LN
54                    1215 NEWTON RD
55                     2605 MCCOY LN
56                1001 NETHERLAND LN
57                 2620 S MADISON ST
58               1410 S CAROLINA AVE
59               1411 S CAROLINA AVE
60               1412 S CAROLINA AVE
61               1413 S CAROLINA AVE
62               1414 S CAROLINA AVE
63               1415 S CAROLINA AVE
64               1415 S CAROLINA AVE
65               1416 S CAROLINA AVE
66                   1330 MOBILE AVE
67            1307 TOWERING PINES LN
68                     302 WELLS AVE
69          2514 WHISPERING PINES RD
70                     701 HOBSON ST
71                 504 S MAGNOLIA ST
72                1521 CORA JINKS LN
73                1110 W WADDELL AVE
74                     1605 SUNNY LN
75                     1609 SUNNY LN
76                    1407 E 3RD AVE
77                   1007 E TIFT AVE
78                  1127 WINGATE AVE
79                      900 20TH AVE
80                 2700 DONCASTER DR
81                 2701 DONCASTER DR
82                  1509 ALABAMA AVE
83                    326 CHERRY AVE
84                   529 COCHRAN AVE
85                   1812 AVALON AVE
86              2226 OLD DOMINION RD
87                 2701 DONCASTER DR
88                250 BONNY VIEW AVE
89                249 BONNY VIEW AVE
90                248 BONNY VIEW AVE
91                251 BONNY VIEW AVE
92                253 BONNY VIEW AVE
93                247 BONNY VIEW AVE
94                254 BONNY VIEW AVE
95                243 BONNY VIEW AVE
96                245 BONNY VIEW AVE
97                252 BONNY VIEW AVE
98                 2515 BRIERWOOD DR
99                  1571 GEORGIA AVE
100              323 INDEPENDENCE DR
101                2700 DONCASTER DR
102             1811 MISSISSIPPI AVE
103                   1411 OWENS AVE
104                    210 WELLS AVE
105                   2415 DAWSON RD
106             525 DON CUTLER SR DR
107                  839 DORSETT AVE
108 

In [37]:
df1_original.head()

Parcel ID                 Address    City State  \
0  0000Q/00025/009     1707 Edgerly Avenue  Albany    GA   
1  0000U/00006/002           703 Dartmouth  Albany    GA   
2  0000H/00047/005  1115 W. Waddell Avenue  Albany    GA   
3  0000X/00003/034           313 Forrestal  Albany    GA   
4  00240/00003/024         1125 Joel Drive  Albany    GA   

            XY Coordinates  Plan Year  IDIS Project                   Project  \
0   31.5735022, -84.191293       2007             6  Homeowner Rehabilitation   
1  31.5882777, -84.1367938       2007             6  Homeowner Rehabilitation   
2  31.5669637, -84.1782082       2007             6  Homeowner Rehabilitation   
3  31.5785109, -84.0824103       2007             6  Homeowner Rehabilitation   
4  31.5611455, -84.2078905       2007             6  Homeowner Rehabilitation   

   IDIS Activity ID Activity Status Program  Funded Amount  Draw Amount  \
0               618       Completed    HOME       27033.46     27033.46   
1               634       Completed    HOME       30185.60     30185.60   
2               610       Completed    HOME       28678.19     28678.19   
3               633       Completed    HOME       34694.47     34694.47   
4               600       Completed    HOME       30756.96     30756.96   

   Balance  
0      0.0  
1      0.0  
2      0.0  
3      0.0  
4      0.0

[0        NA
 1        NA
 2        NA
 3        NA
 4        NA
 5        NA
 6        NA
 7        NA
 8        NA
 9        NA
 10       NA
 11       NA
 12       NA
 13       NA
 14       NA
 15       NA
 16       NA
 17       NA
 18       NA
 19       NA
 20       NA
 21       NA
 22       NA
 23       NA
 24       NA
 25       NA
 26       NA
 27       NA
 28       NA
 29       NA
 30       NA
 31       NA
 32       NA
 33       NA
 34       NA
 35       NA
 36       NA
 37       NA
 38       NA
 39       NA
 40       NA
 41       NA
 42       NA
 43       NA
 44       NA
 45       NA
 46       NA
 47       NA
 48       NA
 49       NA
 50       NA
 51       NA
 52       NA
 53       NA
 54       NA
 55       NA
 56       NA
 57       NA
 58       NA
 59       NA
 60       NA
 61       NA
 62       NA
 63       NA
 64       NA
 65       NA
 66       NA
 67       NA
 68       NA
 69       NA
 70       NA
 71       NA
 72       NA
 73       NA
 74       NA
 75       NA
 76       NA

In [58]:
def appendaddress(list_series):
    tmp_to_series = []
    for (apt, dup, sno, snm, stp) in zip(list_series[0],list_series[1],list_series[2],
                                         list_series[3],list_series[4]):
        word = ""
        if sno != "NA":
            word += sno + " "
        #if dup != "NA":
        #    word += dup
        word += snm + " "
        if stp != "NA":
            word += stp + " "
        #if apt != "NA":
        #    word += "APT " + apt
        word = word.strip()
        tmp_to_series.append(word)
    return pd.Series(tmp_to_series)

In [63]:
def appendmulti(list_series):
    tmp_to_series = []
    for (apt, dup, sno, snm, stp) in zip(list_series[0],list_series[1],list_series[2],
                                         list_series[3],list_series[4]):
        word = ""
        #if sno != "NA":
        #    word += sno + " "
        if dup != "NA":
            word += dup
        #word += snm + " "
        #if stp != "NA":
        #    word += stp + " "
        elif apt != "NA":
            word += apt
        else:
            word = "NA"
        word = word.strip()
        tmp_to_series.append(word)
    return pd.Series(tmp_to_series)   

In [67]:
def appendapt(list_series):
    tmp_to_series = []
    for (apt, dup, sno, snm, stp) in zip(list_series[0],list_series[1],list_series[2],
                                         list_series[3],list_series[4]):
        word = ""
        #if sno != "NA":
        #    word += sno + " "
        #if dup != "NA":
        #    word += dup
        #word += snm + " "
        #if stp != "NA":
        #    word += stp + " "
        if apt != "NA":
            word += "APT"
        else:
            word = "NA"
        word = word.strip()
        tmp_to_series.append(word)
    return pd.Series(tmp_to_series)   

In [92]:
main_addr = appendaddress(home)
df1_original['Address'] = main_addr

In [93]:
apt_addr = appendapt(home)
df1_original['Multihome_Type'] = apt_addr

In [94]:
multi_addr = appendmulti(home)
df1_original['Multihome'] = multi_addr

In [95]:
main_addr_cdbg = appendaddress(cdbg)
multi_addr_cdbg = appendmulti(cdbg)
apt_addr_cdbg = appendapt(cdbg)

In [96]:
df2_original['Address'] = main_addr_cdbg
df2_original['Multihome_Type'] = apt_addr_cdbg
df2_original['Multihome'] = multi_addr_cdbg

In [66]:
multi_addr

0        NA
1        NA
2        NA
3        NA
4        NA
5        NA
6        NA
7        NA
8        NA
9        NA
10       NA
11       NA
12       NA
13       NA
14       NA
15       NA
16       NA
17       NA
18       NA
19       NA
20       NA
21       NA
22       NA
23       NA
24       NA
25       NA
26       NA
27       NA
28       NA
29       NA
30       NA
31       NA
32       NA
33       NA
34       NA
35       NA
36       NA
37       NA
38       NA
39       NA
40       NA
41       NA
42       NA
43       NA
44       NA
45       NA
46       NA
47       NA
48       NA
49       NA
50       NA
51       NA
52       NA
53       NA
54       NA
55       NA
56       NA
57       NA
58       NA
59       NA
60       NA
61       NA
62       NA
63       NA
64       NA
65       NA
66       NA
67       NA
68       NA
69       NA
70       NA
71       NA
72       NA
73       NA
74       NA
75       NA
76       NA
77       NA
78       NA
79       NA
80        A
81        A
82       NA
83  

In [52]:
df1_original

Parcel ID                        Address    City State  \
0         0000Q/00025/009               1707 EDGERLY AVE  Albany    GA   
1         0000U/00006/002               703 DARTMOUTH LN  Albany    GA   
2         0000H/00047/005             1115 W WADDELL AVE  Albany    GA   
3         0000X/00003/034               313 FORRESTAL LN  Albany    GA   
4         00240/00003/024                   1125 JOEL DR  Albany    GA   
5         0000K/00027/008                 1519 MALONE DR  Albany    GA   
6         0000U/00003/010                 719 ANDOVER LN  Albany    GA   
7         0000Q/00026/007            305 S SHADOWLAWN DR  Albany    GA   
8         000HH/00062/042                  1662 SUNNY LN  Albany    GA   
9         00001/00031/24D                511 MOULTRIE RD  Albany    GA   
10        000TT/00003/02A                  1100 SWIFT ST  Albany    GA   
11        000EE/00013/005           1009 E ROOSEVELT AVE  Albany    GA   
12        0000H/00018/013             1118 W WHITNEY AVE  Albany    GA   
13        000BB/00015/642                   431 EBONY LN  Albany    GA   
14        000BB/00015/619                   529 EBONY LN  Albany    GA   
15        000BB/00015/616                   535 EBONY LN  Albany    GA   
16        000BB/00015/614                   539 EBONY LN  Albany    GA   
17        000BB/00015/611                   545 EBONY LN  Albany    GA   
18        000BB/00015/604                   559 EBONY LN  Albany    GA   
19        000BB/00015/602                   563 EBONY LN  Albany    GA   
20        000BB/00015/618                   531 EBONY LN  Albany    GA   
21        000BB/00015/615                   537 EBONY LN  Albany    GA   
22        000BB/00015/612                   543 EBONY LN  Albany    GA   
23        000BB/00015/606                   555 EBONY LN  Albany    GA   
24        0000W/00003/006             1403 GREENBRIER CT  Albany    GA   
25        000NN/00013/006                 1802 SHARON DR  Albany    GA   
26        00012/00006/30A                533 AMERICAN LN  Albany    GA   
27        00012/00006/026                539 AMERICAN LN  Albany    GA   
28        0000K/00017/008                1602 ESTELLE ST  Albany    GA   
29        000BB/00014/01O                 456 GAINES AVE  Albany    GA   
30        000HH/00023/010                  621 BURKE AVE  Albany    GA   
31        00230/00010/002                 1910 JONES AVE  Albany    GA   
32        000GG/00031/004                 514 EUGEMAR DR  Albany    GA   
33        000BB/00014/01O                 456 GAINES AVE  Albany    GA   
34        000BB/00014/01P                 460 GAINES AVE  Albany    GA   
35        000BB/00014/01P                 460 GAINES AVE  Albany    GA   
36        00365/00005/019            616 INDIAN CREEK DR  Albany    GA   
37        0000I/00013/005  1005 MARTIN LUTHER KING JR DR  Albany    GA   
38        0000K/00045/030                 1411 OWENS AVE  Albany    GA   
39        000BB/00029/016           2230 S WASHINGTON ST  Albany    GA   
40        0000I/00013/002            1006 S JEFFERSON ST  Albany    GA   
41        0000I/00008/018                 402 COTTON AVE  Albany    GA   
42        0000K/00045/026                 1403 OWENS AVE  Albany    GA   
43        0000I/00033/048               426 ROBINSON AVE  Albany    GA   
44        000HH/00042/007                1212 S DAVIS ST  Albany    GA   
45        000HH/00066/037                1203 HAMPTON CT  Albany    GA   
46        000HH/00066/034                1209 HAMPTON CT  Albany    GA   
47        00017/00000/064                  118 ALMOND DR  Albany    GA   
48        000HH/00066/030                1305 HAMPTON CT  Albany    GA   
49        000HH/00066/030                1305 HAMPTON CT  Albany    GA   
50        00012/00006/041                509 AMERICAN LN  Albany    GA   
51        00012/00006/036                519 AMERICAN LN  Albany    GA   
52        00186/00003/051                1324 MOBILE AVE  Albany    GA   
53        00186/0000

In [69]:
apt_addr

0       NA
1       NA
2       NA
3       NA
4       NA
5       NA
6       NA
7       NA
8       NA
9       NA
10      NA
11      NA
12      NA
13      NA
14      NA
15      NA
16      NA
17      NA
18      NA
19      NA
20      NA
21      NA
22      NA
23      NA
24      NA
25      NA
26      NA
27      NA
28      NA
29      NA
30      NA
31      NA
32      NA
33      NA
34      NA
35      NA
36      NA
37      NA
38      NA
39      NA
40      NA
41      NA
42      NA
43      NA
44      NA
45      NA
46      NA
47      NA
48      NA
49      NA
50      NA
51      NA
52      NA
53      NA
54      NA
55      NA
56      NA
57      NA
58      NA
59      NA
60      NA
61      NA
62      NA
63      NA
64      NA
65      NA
66      NA
67      NA
68      NA
69      NA
70      NA
71      NA
72      NA
73      NA
74      NA
75      NA
76      NA
77      NA
78      NA
79      NA
80      NA
81      NA
82      NA
83      NA
84      NA
85      NA
86      NA
87      NA
88      NA
89      NA
90      NA

In [97]:
df1_original

Parcel ID                        Address    City State  \
0         0000Q/00025/009               1707 EDGERLY AVE  Albany    GA   
1         0000U/00006/002               703 DARTMOUTH LN  Albany    GA   
2         0000H/00047/005             1115 W WADDELL AVE  Albany    GA   
3         0000X/00003/034               313 FORRESTAL LN  Albany    GA   
4         00240/00003/024                   1125 JOEL DR  Albany    GA   
5         0000K/00027/008                 1519 MALONE DR  Albany    GA   
6         0000U/00003/010                 719 ANDOVER LN  Albany    GA   
7         0000Q/00026/007            305 S SHADOWLAWN DR  Albany    GA   
8         000HH/00062/042                  1662 SUNNY LN  Albany    GA   
9         00001/00031/24D                511 MOULTRIE RD  Albany    GA   
10        000TT/00003/02A                  1100 SWIFT ST  Albany    GA   
11        000EE/00013/005           1009 E ROOSEVELT AVE  Albany    GA   
12        0000H/00018/013             1118 W WHITNEY AVE  Albany    GA   
13        000BB/00015/642                   431 EBONY LN  Albany    GA   
14        000BB/00015/619                   529 EBONY LN  Albany    GA   
15        000BB/00015/616                   535 EBONY LN  Albany    GA   
16        000BB/00015/614                   539 EBONY LN  Albany    GA   
17        000BB/00015/611                   545 EBONY LN  Albany    GA   
18        000BB/00015/604                   559 EBONY LN  Albany    GA   
19        000BB/00015/602                   563 EBONY LN  Albany    GA   
20        000BB/00015/618                   531 EBONY LN  Albany    GA   
21        000BB/00015/615                   537 EBONY LN  Albany    GA   
22        000BB/00015/612                   543 EBONY LN  Albany    GA   
23        000BB/00015/606                   555 EBONY LN  Albany    GA   
24        0000W/00003/006             1403 GREENBRIER CT  Albany    GA   
25        000NN/00013/006                 1802 SHARON DR  Albany    GA   
26        00012/00006/30A                533 AMERICAN LN  Albany    GA   
27        00012/00006/026                539 AMERICAN LN  Albany    GA   
28        0000K/00017/008                1602 ESTELLE ST  Albany    GA   
29        000BB/00014/01O                 456 GAINES AVE  Albany    GA   
30        000HH/00023/010                  621 BURKE AVE  Albany    GA   
31        00230/00010/002                 1910 JONES AVE  Albany    GA   
32        000GG/00031/004                 514 EUGEMAR DR  Albany    GA   
33        000BB/00014/01O                 456 GAINES AVE  Albany    GA   
34        000BB/00014/01P                 460 GAINES AVE  Albany    GA   
35        000BB/00014/01P                 460 GAINES AVE  Albany    GA   
36        00365/00005/019            616 INDIAN CREEK DR  Albany    GA   
37        0000I/00013/005  1005 MARTIN LUTHER KING JR DR  Albany    GA   
38        0000K/00045/030                 1411 OWENS AVE  Albany    GA   
39        000BB/00029/016           2230 S WASHINGTON ST  Albany    GA   
40        0000I/00013/002            1006 S JEFFERSON ST  Albany    GA   
41        0000I/00008/018                 402 COTTON AVE  Albany    GA   
42        0000K/00045/026                 1403 OWENS AVE  Albany    GA   
43        0000I/00033/048               426 ROBINSON AVE  Albany    GA   
44        000HH/00042/007                1212 S DAVIS ST  Albany    GA   
45        000HH/00066/037                1203 HAMPTON CT  Albany    GA   
46        000HH/00066/034                1209 HAMPTON CT  Albany    GA   
47        00017/00000/064                  118 ALMOND DR  Albany    GA   
48        000HH/00066/030                1305 HAMPTON CT  Albany    GA   
49        000HH/00066/030                1305 HAMPTON CT  Albany    GA   
50        00012/00006/041                509 AMERICAN LN  Albany    GA   
51        00012/00006/036                519 AMERICAN LN  Albany    GA   
52        00186/00003/051                1324 MOBILE AVE  Albany    GA   
53        00186/0000

In [98]:
df2_original

Parcel ID                            Address    City State  \
0          NaN               1008 E RESIDENCE AVE  Albany    GA   
1          NaN                      1100 SWIFT ST  Albany    GA   
2          NaN                 1117 W WADDELL AVE  Albany    GA   
3          NaN                   1223 E BROAD AVE  Albany    GA   
4          NaN                     1316 E 4TH AVE  Albany    GA   
5          NaN                1419 W HIGHLAND AVE  Albany    GA   
6          NaN                    1607 AVALON AVE  Albany    GA   
7          NaN                      1662 SUNNY LN  Albany    GA   
8          NaN                1706 S JEFFERSON ST  Albany    GA   
9          NaN                1712 W HIGHLAND AVE  Albany    GA   
10         NaN                  1805 S JACKSON ST  Albany    GA   
11         NaN                  1805 S JACKSON ST  Albany    GA   
12         NaN                     1904 JONES AVE  Albany    GA   
13         NaN                1925 S RIVERVIEW DR  Albany    GA   
14         NaN                  2002 N SKYLANE DR  Albany    GA   
15         NaN                 207 N INGLESIDE DR  Albany    GA   
16         NaN                    2109 JUNIPER DR  Albany    GA   
17         NaN                  2126 COLQUITT AVE  Albany    GA   
18         NaN                      2201 GRADY ST  Albany    GA   
19         NaN                  2205 EVERGREEN DR  Albany    GA   
20         NaN                      2215 GRADY ST  Albany    GA   
21         NaN                        234 CONE ST  Albany    GA   
22         NaN                  2500 HABERSHAM RD  Albany    GA   
23         NaN                  2502 HABERSHAM RD  Albany    GA   
24         NaN                305 S SHADOWLAWN DR  Albany    GA   
25         NaN                322 INDEPENDENCE DR  Albany    GA   
26         NaN                     3400 NEWTON RD  Albany    GA   
27         NaN                     3400 NEWTON RD  Albany    GA   
28         NaN                     3404 NEWTON RD  Albany    GA   
29         NaN                     3404 NEWTON RD  Albany    GA   
30         NaN                       608 AZTEC LN  Albany    GA   
31         NaN                  624 JEFFERIES AVE  Albany    GA   
32         NaN                     629 HARMON AVE  Albany    GA   
33         NaN                     630 GAINES AVE  Albany    GA   
34         NaN                     637 HARMON AVE  Albany    GA   
35         NaN                     639 HARMON AVE  Albany    GA   
36         NaN                     639 HARMON AVE  Albany    GA   
37         NaN                     640 HARMON AVE  Albany    GA   
38         NaN                   701 BARKLEY BLVD  Albany    GA   
39         NaN                       705 16TH AVE  Albany    GA   
40         NaN                823 S SHADOWLAWN DR  Albany    GA   
41         NaN                    UNKNOWN ADDRESS  Albany    GA   
42         NaN                801 W RESIDENCE AVE  Albany    GA   
43         NaN                   230 S JACKSON ST  Albany    GA   
44         NaN                 314 N JEFFERSON ST  Albany    GA   
45         NaN                 603 N JEFFERSON ST  Albany    GA   
46         NaN                   230 S JACKSON ST  Albany    GA   
47         NaN                    835 W BROAD AVE  Albany    GA   
48         NaN                       400 EBONY LN  Albany    GA   
49         NaN                     101 S FRONT ST  Albany    GA   
50         NaN                 638 W HIGHLAND AVE  Albany    GA   
51         NaN                    UNKNOWN ADDRESS  Albany    GA   
52         NaN                    UNKNOWN ADDRESS  Albany    GA   
53         NaN                1601 S SLAPPEY BLVD  Albany    GA   
54         NaN                    UNKNOWN ADDRESS  Albany    GA   
55         NaN                      300 W 3RD AVE  Albany    GA   
56         NaN                      1003 SWIFT ST  Albany    GA   
57         NaN                   1106 CRAWFORD DR  Albany    GA   
58         NaN                 1118 W WHITNEY AVE  A

In [15]:
df = pd.concat([df1_original, df2_original], ignore_index = True)

In [103]:
df.to_csv('~/Dropbox (Amherst College)/CDS-2019-AlbanyHub/ToDatabase/TotalHome.csv', index = False)

In [25]:
s_house_addresses = pd.Series(df_houses['Address'].unique())

In [26]:
s_addresses

0                      212 ACORN ST
1                  2101 LEONARD AVE
2                  2103 LEONARD AVE
3                 636 JEFFERIES AVE
4                  2105 LEONARD AVE
5                      201 ELSOM ST
6                      203 ELSOM ST
7                      205 ELSOM ST
8                      207 ELSOM ST
9                      209 ELSOM ST
10                  2214 HUBERT AVE
11                     301 ELSOM ST
12                     303 ELSOM ST
13                640 JEFFERIES AVE
14                     305 ELSOM ST
15                     309 ELSOM ST
16                     313 ELSOM ST
17                     314 ELSOM ST
18                     306 ELSOM ST
19                     300 ELSOM ST
20                   1504 NEWTON RD
21                  2305 HUBERT AVE
22                  2309 HUBERT AVE
23                 2200 LEONARD AVE
24                  109 N MOCK N RD
25                    107 N MOCK RD
26                   1506 NEWTON RD
27                   2100 CL

In [30]:
tmp = s_house_addresses[~s_house_addresses.isin(s_addresses)]

In [32]:
tmp.to_csv('~/Dropbox (Amherst College)/CDS-2019-AlbanyHub/Raw-Data/mismatched_house_addresses.csv', header = False)

In [24]:
df_houses.head()

Parcel ID             Address    City State           XY Coordinates  \
0  0000Q/00025/009    1707 EDGERLY AVE  Albany    GA   31.5735022, -84.191293   
1  0000U/00006/002    703 DARTMOUTH LN  Albany    GA  31.5882777, -84.1367938   
2  0000H/00047/005  1115 W WADDELL AVE  Albany    GA  31.5669637, -84.1782082   
3  0000X/00003/034    313 FORRESTAL LN  Albany    GA  31.5785109, -84.0824103   
4  00240/00003/024        1125 JOEL DR  Albany    GA  31.5611455, -84.2078905   

   Plan Year  IDIS Project                   Project  IDIS Activity ID  \
0       2007             6  Homeowner Rehabilitation               618   
1       2007             6  Homeowner Rehabilitation               634   
2       2007             6  Homeowner Rehabilitation               610   
3       2007             6  Homeowner Rehabilitation               633   
4       2007             6  Homeowner Rehabilitation               600   

  Activity Status Program  Funded Amount  Draw Amount  Balance Multihome_Type  \
0       Completed    HOME       27033.46     27033.46      0.0            NaN   
1       Completed    HOME       30185.60     30185.60      0.0            NaN   
2       Completed    HOME       28678.19     28678.19      0.0            NaN   
3       Completed    HOME       34694.47     34694.47      0.0            NaN   
4       Completed    HOME       30756.96     30756.96      0.0            NaN   

  Multihome  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN